packages/react-i18n/src/index.tsx

```diff
@@ -16,17 +16,51 @@ export interface I18nReact {
 
 const I18nContext = React.createContext< I18nReact >( makeContextValue() );
 
+export type ChangeLocaleFunction = ( newLocale: string ) => void;
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const ChangeLocaleContext = React.createContext< ChangeLocaleFunction >( () => {} );
+
 interface Props {
-	localeData?: LocaleData;
+	initialLocaleData?: LocaleData;
+	getLanguageFile: ( targetLocaleSlug: string ) => Promise< LocaleData >;
+	defaultLocaleSlug: string;
 }
-export const I18nProvider: React.FunctionComponent< Props > = ( { children, localeData } ) => {
+
+export const I18nProvider: React.FunctionComponent< Props > = ( {
```

Comment:
@sirreal does this make sense to you to expand this I18nProvider class to also include changing locales? It required me to pass in some extra parameters, especially the `getLanguageFile` function. I was thinking it might even make sense to share this with calypso eventually? related to #42997

Reply:
This might make sense eventually, but I'd like to leave this package as is and implement locale changing in Calypso application code. The package is quite simple and focused on providing the translation functionality, I'm reluctant to have it start managing more things.

Reply:
ok, that makes sense 👍